mutation/index.ts

```diff
@@ -0,0 +1,125 @@
+import { useCallback, useRef } from 'react'
+import useSWR, { useSWRConfig, Middleware, Key } from 'swr'
+
+import { serialize } from '../src/utils/serialize'
+import { useStateWithDeps } from '../src/utils/state'
+import { withMiddleware } from '../src/utils/with-middleware'
+import { useIsomorphicLayoutEffect } from '../src/utils/env'
+import { UNDEFINED } from '../src/utils/helper'
+import { getTimestamp } from '../src/utils/timestamp'
+
+import {
+  SWRMutationConfiguration,
+  SWRMutationResponse,
+  SWRMutationHook,
+  MutationFetcher
+} from './types'
+
+const mutation =
+  <Data, Error>() =>
+  (
+    key: Key,
+    fetcher: MutationFetcher<Data>,
+    config: SWRMutationConfiguration<Data, Error> = {}
+  ) => {
+    const { mutate } = useSWRConfig()
+
+    const keyRef = useRef(key)
+    // Ditch all mutation results that happened earlier than this timestamp.
+    const ditchMutationsTilRef = useRef(0)
+
+    const [stateRef, stateDependencies, setState] = useStateWithDeps(
+      {
+        data: UNDEFINED,
+        error: UNDEFINED,
+        isMutating: false
+      },
+      true
+    )
+    const currentState = stateRef.current
+
+    // Similar to the global mutate, but bound to the current cache and key.
+    // `cache` isn't allowed to change during the lifecycle.
+    const boundMutate = useCallback(
+      (arg0, arg1) => mutate(serialize(keyRef.current)[0], arg0, arg1),
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      []
+    )
+
+    const trigger = useCallback(async (extraArg, opts) => {
+      if (!fetcher) {
+        throw new Error('Can’t trigger the mutation: missing fetcher.')
+      }
+
+      const [serializedKey, args] = serialize(keyRef.current)
+
+      // Disable cache population by default.
+      const options = Object.assign({ populateCache: false }, config, opts)
+
+      // Trigger a mutation, also track the timestamp. Any mutation that happened
+      // earlier this timestamp should be ignored.
+      const mutationStartedAt = getTimestamp()
+      ditchMutationsTilRef.current = mutationStartedAt
+
+      setState({ isMutating: true })
+      args.push(extraArg)
+
+      try {
+        const data = await mutate(
+          serializedKey,
+          (fetcher as any)(...args),
```

Comment:
How about deleting line 65 and writing here like this:

```tsx
(fetcher as any)(...args.concat([extraArg])),
```

Reply:
Good call, thanks.